writeengine/wrapper/writeengine.cpp

```diff
@@ -4280,27 +4590,47 @@ int WriteEngineWrapper::updateColumnRec(const TxnID& txnid,
                 {
                     RETURN_ON_ERROR(AddLBIDtoList(txnid,
                                                   colStructList[j],
-                                                  curFbo,
-						  dummyCPInfo));
+                                                  curFbo, cpInfoP));
                 }
             }
         }
 
 //#ifdef PROFILE
 //timer.start("markExtentsInvalid");
 //#endif
-        markTxnExtentsAsInvalid(txnid);
 
         if (m_opType != DELETE)
             m_opType = UPDATE;
 
-        rc = writeColumnRec(txnid, cscColTypeList, colStructList, colValueList, colOldValueList,
-                            ridLists[extent], tableOid, true, ridLists[extent].size());
+        rc = writeColumnRecUpdate(txnid, cscColTypeList, colStructList, colValueList, colOldValueList,
+                            ridLists[extent], tableOid, true, ridLists[extent].size(), &currentExtentRangesPtrs);
 
-        m_opType = NOOP;
+//        m_opType = NOOP; // XXX: WHY WAS IT HERE???!!!
 
         if (rc != NO_ERROR)
             break;
+
+	// copy updated ranges into bulk update vector.
+	for(auto cpInfoPtr : currentExtentRangesPtrs)
+	{
+	    if (cpInfoPtr)
+	    {
+		cpInfoPtr->seqNum ++;
+	        infosToUpdate.push_back(*cpInfoPtr);
+	    }
+	}
+    }
+    markTxnExtentsAsInvalid(txnid);
+    if (rc == NO_ERROR)
+    {
+        std::vector<BRM::CPInfo> infosToDrop = infosToUpdate;
+	for (auto& cpInfo : infosToDrop)
```

Comment:
Not sure why but some of the lines are totaly misaligned.

Reply:
Forgot tabs->spaces.